adhocracy4/static/api.js

```diff
@@ -73,6 +85,30 @@ const api = (function () {
     return $.ajax(params)
   }
 
+  // This allows to use api-urls directly
+  // In contrast to _sendRequest which is used for more
+  // complex requests (e.g. constructs its api-urls itself)
+  function _sendDirectRequest (url, options) {
+    const $body = $('body')
+
+    if (options && options.params) {
+      url = url + encodeQueryString(options.params)
+    }
+
+    const defaultParams = {
+      url: url,
+      error: function (xhr, status, err) {
+        console.error(url, status, err.toString())
+      },
+      complete: function () {
+        $body.removeClass('loading')
+      }
+    }
+
+    $body.addClass('loading')
+    return $.ajax(defaultParams)
+  }
+
```

Comment:
Would there also be a way to use the other _sendRequest? I find it a bit confusing that we now here introduce another function, whereas we use _sendRequest for posting/changing comments.. But probably I dont get it? Happy to discuss!

Reply:
as we would refactor this file at some point to get rid of jQuery's ajax() I thought it would be easier to "duplicate" this method for now.

Actually I was thinking either to pass the same request infos (e.g. baseUrl + contenttypeid + comment id) to use the _sendRequest. But that seems complicated.

The other way would be to make sure _sendRequest can work with api-urls directly but that would also complicate the _sendRequest. So I thought making its own method is the easiest way, no? @Rineee

Reply:
Yeah, no I think make _sendRequest work with api-urls directly does not really make sense.

I see your point that it is easier this ways, but tbh I would prefer using it the same way as with all other requests, because then we could also get rid of the commentApiUrl that is passed through template tag. And also for refactoring isnt it easier if we just have one way of sending requests and passing infos?

I can also look into it if you want!